### ISSUE TITLE
scipy-notebook: pandas 0.22.0, {...}

### ISSUE THREAD

**Issue by westurner** [jupyter/docker-stacks#530]
http://pandas.pydata.org/pandas-docs/stable/

http://pandas.pydata.org/pandas-docs/stable/whatsnew.html

https://github.com/jupyter/docker-stacks/blob/master/scipy-notebook/Dockerfile

**Comment by parente** [comment]
@westurner Thanks. Happy to take a PR to update any / all libraries.

**Comment by Bidek56** [comment]
You can just update the libraries in a local Dockerfile, no need to wait for the base update.

###### Dockerfile ######
FROM jupyter/pyspark-notebook:latest
USER root
RUN conda update -y  --all

**Comment by westurner** [comment]
Yup, but then I download the image containing the old versions and also the
updated conda packages.

`conda update -y all; conda env export` should list the latest package
versions.

Without tests, I'd assume that the latest versions are all working?

**Comment by westurner** [comment]
Is there a reason not to manage these versions in an external environment.yml?

- environment.yml would be easier to diff and patch separately (``conda env export -f environment.yml``)
- It's easy to see the versions in the one Dockerfile

**Comment by kcolton** [comment]
@westurner 
> Without tests, I'd assume that the latest versions are all working?
Without tests, I'de assume nothing about versions working. 

Version bumps need to be done methodically and care taken to look at what breaking changes. So that the next person who happens to do a `docker pull jupyter/*-notebook:latest` might find their notebooks suddenly broken.

I need a feature from `pandas=0.22.0` so i've done some above overrides. 

It would be great to allow everyone else to enjoy the new features "out of the box."

@parente Forking to do the bump while also keeping in mind:
https://github.com/jupyter/docker-stacks/pull/562
https://github.com/jupyter/docker-stacks/pull/522

Will get the tests running, look at the breaking changes just for 0.22 and, possible conflicts with the other PR versions and hopefully submit a PR this weekend.

With all the updates around Jupyter (labs, lots of extension points, k8s) do you know what the roadmap is for stacks? 

Happy to look into other version bumps and add docs about new features in versions if this repo and approach is going to stick around.

**Comment by parente** [comment]
The mantra for this repo has always been that users should be using specific build tags for the images here, unless they're comfortable with anything changing under their feet with `latest`. The README calls this out.

We have to have some measure of trust that pandas 0.22 has been tested by the pandas community and is not completely broken. That said, there barebones automation for running functional tests in pytest in place here. Right now, it's used to test notebook liveliness after a build and various options in the base-notebook image (#508). I think it's possible to use nbconvert + the execute preprocessor to run some test notebooks as part of the build process if someone would like to contribute on that front.

Long-term, #517 is the start of a conversation about how to reduce the maintenance burden here and help people setup their own Docker build process (e.g., starting from some documentation #538 + a cookiecutter github project).

**Comment by westurner** [comment]
> I think it's possible to use nbconvert + the execute preprocessor to run some test notebooks as part of the build process if someone would like to contribute on that front.

http://nbconvert.readthedocs.io/en/latest/execute_api.html

```bash
jupyter nbconvert --to notebook --execute mynotebook.ipynb
```

https://github.com/computationalmodelling/nbval

> **Py.test plugin for validating Jupyter notebooks**
>
> The plugin adds functionality to py.test to recognise and collect Jupyter notebooks. The intended purpose of the tests is to determine whether execution of the stored inputs match the stored outputs of the .ipynb file. Whilst also ensuring that the notebooks are running without errors.

```
py.test --nbval
py.test --nbval-lax
py.test --nbval --sanitize-with path/to/my_sanitize_file
```

**Comment by westurner** [comment]
Thanks!
